pkg/operator/operator.go

```diff
@@ -174,11 +179,31 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		launchTemplateProvider,
 	)
 
+	//v2
+	cfg, err := configV2.LoadDefaultConfig(ctx,
+		configV2.WithRetryMaxAttempts(5),
+	)
+	if err != nil {
+		panic(err)
+	}
+	prometheusv2.WithPrometheusMetrics(cfg, crmetrics.Registry)
+	if cfg.Region == "" {
+		log.FromContext(ctx).V(1).Info("retrieving region from IMDS")
+		metaDataClient := imds.NewFromConfig(cfg)
+		region, err := metaDataClient.GetRegion(ctx, nil)
+		if err != nil {
+			log.FromContext(ctx).Error(err, "failed to get region from metadata server")
+			os.Exit(1)
+		}
+		cfg.Region = region.Region
+	}
+
+	instanceProfileProvider := instanceprofile.NewDefaultProvider(cfg.Region, iamV2.NewFromConfig(cfg), cache.New(awscache.InstanceProfileTTL, awscache.DefaultCleanupInterval))
```

Comment:
Can we leave this one the line it was defined before? Trying to think the best way to structure this files would be